neqo-http3/src/hframe.rs

```diff
@@ -37,14 +38,116 @@ pub enum HStreamType {
     Push,
 }
 
-#[derive(PartialEq, Debug)]
+#[derive(Clone, PartialEq, Debug, Copy)]
 pub enum HSettingType {
     MaxHeaderListSize,
-    MaxTableSize,
+    MaxTableCapacity,
     BlockedStreams,
     UnknownType,
 }
 
+fn hsetting_default(setting_type: HSettingType) -> u64 {
+    match setting_type {
+        HSettingType::MaxHeaderListSize => 1 << 62,
+        _ => 0,
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct HSetting {
+    pub setting_type: HSettingType,
+    pub value: u64,
+}
+
+impl HSetting {
+    pub fn new(setting_type: HSettingType, value: u64) -> Self {
+        HSetting {
+            setting_type,
+            value,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct HSettings {
+    settings: Vec<HSetting>,
+}
+
+impl HSettings {
+    pub fn new(settings: Vec<HSetting>) -> Self {
```

Comment:
This should take a slice and not a Vec.  That will make some of the code below in tests slightly easier to work with.